### ISSUE TITLE
Class cast exception with OSATE+BA from the update site

### ISSUE THREAD

**Issue by Etienne13** [osate/osate2-ba#35]
When using OSATE+BA with a command line interface, I have a class cast exception in the name resolver of the BA plugin.

This issues appeared with binaries from version 2.2.2 (both the stable and testing update sites). Binaries of version 2.2.1 work fine.

Unfortunately, I am not able to reproduce the bug when using the source code of OSATE+BA, neither with source code of the master nor with source code from the develop branch.

Here is the result of my tests, when using different release of OSATE+BA:
http://aadl.info/aadl/osate/stable/2.2.1/update-site/
--> The issue does no appear

http://aadl.info/aadl/osate/stable/latest/update-site/
--> ClassCastException

http://aadl.info/aadl/osate/testing/update-site/
--> ClassCastException

Testing with the product (not the update site): http://aadl.info/aadl/osate/stable/latest/products/
—> The issue does not appear.

Considering the  the exception trace bellow and the commit d5e2e98352dfef675c829ab8d075aba940f2dda0 (https://github.com/osate/osate2-core/commit/d5e2e98352dfef675c829ab8d075aba940f2dda0#diff-c778da4cb37b2e32311a9b9b41798b95) even though it is 1 year old, my guess is that the build of the BA on the update site is based on an old version of osate source code. Indeed, when using the BA from the update site I could notice when debuting that the class identifier returned by method getClassifierID was not consistent with the identifiers in the Aadl2Package class.

Looking at the build numbers for BA in http://aadl.info/aadl/osate/stable/latest/update-site/plugins/ it seems they are quite old. Maybe a re-compilation of the BA plugin would solve the issue...?

As I have a nightly build with the latest version on the update site I should be able to detect if it solves the problem.

Thanks!
Etienne.

java.lang.ClassCastException: org.osate.aadl2.impl.ListValueImpl cannot be cast to org.osate.aadl2.NamedValue
	at org.osate.utils.PropertyUtils.getValue(PropertyUtils.java:1245)
	at org.osate.ba.analyzers.AadlBaNameResolver.propertyAssociationResolver(AadlBaNameResolver.java:2073)
	at org.osate.ba.analyzers.AadlBaNameResolver.propertyNameResolver(AadlBaNameResolver.java:1825)
	at org.osate.ba.analyzers.AadlBaNameResolver.classifierPropertyReferenceResolver(AadlBaNameResolver.java:1681)
	at org.osate.ba.analyzers.AadlBaNameResolver.propertyReferenceResolver(AadlBaNameResolver.java:1556)
	at org.osate.ba.analyzers.AadlBaNameResolver.valueConstantResolver(AadlBaNameResolver.java:1498)
	at org.osate.ba.analyzers.AadlBaNameResolver.valueResolver(AadlBaNameResolver.java:2356)
	at org.osate.ba.analyzers.AadlBaNameResolver.valueExpressionResolver(AadlBaNameResolver.java:2329)
	at org.osate.ba.analyzers.AadlBaNameResolver.assignmentActionResolver(AadlBaNameResolver.java:167)
	at org.osate.ba.analyzers.AadlBaNameResolver.basicActionResolver(AadlBaNameResolver.java:290)
	at org.osate.ba.analyzers.AadlBaNameResolver.behaviorActionResolver(AadlBaNameResolver.java:464)
	at org.osate.ba.analyzers.AadlBaNameResolver.behaviorActionsResolver(AadlBaNameResolver.java:644)
	at org.osate.ba.analyzers.AadlBaNameResolver.ifStatementResolver(AadlBaNameResolver.java:438)
	at org.osate.ba.analyzers.AadlBaNameResolver.behaviorActionResolver(AadlBaNameResolver.java:477)
	at org.osate.ba.analyzers.AadlBaNameResolver.behaviorActionsResolver(AadlBaNameResolver.java:644)
	at org.osate.ba.analyzers.AadlBaNameResolver.ifStatementResolver(AadlBaNameResolver.java:438)
	at org.osate.ba.analyzers.AadlBaNameResolver.behaviorActionResolver(AadlBaNameResolver.java:477)
	at org.osate.ba.analyzers.AadlBaNameResolver.behaviorActionsResolver(AadlBaNameResolver.java:638)
	at org.osate.ba.analyzers.AadlBaNameResolver.behaviorActionBlockResolver(AadlBaNameResolver.java:399)
	at org.osate.ba.analyzers.AadlBaNameResolver.behaviorTransitionResolver(AadlBaNameResolver.java:391)
	at org.osate.ba.analyzers.AadlBaNameResolver.resolveNames(AadlBaNameResolver.java:1121)
	at org.osate.ba.AadlBaResolver.resolveAnnex(AadlBaResolver.java:65)
	at org.osate.annexsupport.AnnexResolverProxy.resolveAnnex(AnnexResolverProxy.java:70)
	at fr.tpt.aadl.ramses.control.cli.instantiation.AnnexJob.resolve(AnnexJob.java:151)
	at fr.tpt.aadl.ramses.control.cli.instantiation.AnnexJobHandler.resolveAllAnnexes(AnnexJobHandler.java:56)
	at fr.tpt.aadl.ramses.control.cli.instantiation.StandAloneInstantiator.parse(StandAloneInstantiator.java:165)
	at fr.tpt.aadl.ramses.control.cli.instantiation.StandAloneInstantiator.parse(StandAloneInstantiator.java:129)
	at fr.tpt.aadl.ramses.control.cli.core.ToolSuiteLauncher.launchDefaultGenerationProcess(ToolSuiteLauncher.java:183)
	at fr.tpt.aadl.ramses.control.cli.core.ToolSuiteLauncherCommand.defaultProcess(ToolSuiteLauncherCommand.java:588)
	at fr.tpt.aadl.ramses.control.cli.core.ToolSuiteLauncherCommand.main(ToolSuiteLauncherCommand.java:164)
	at fr.tpt.aadl.ramses.control.cli.osgi.Application.start(Application.java:60)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:669)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:608)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1515)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1488)

**Comment by lwrage** [comment]
This is strange...

I don't think it's a build issue:
 - The date on the jar is the date of the last commit that is included in the build and reflects that there is essentially no development activity for the BA plugin. It's not the date of the build.
 - The build is always a full build which compiles everything. 
 - Also, the build process works by first creating the update site and then assembling the product from that update site.
 - I checked, and product and update site for 2.2.2.vfinal really contain the same org.osate.* plugin jars.

As you are getting different behavior from product vs. update site, please check if the plugin versions after installation from the update site match the plugin versions in the installed product. By the way, did you try to update an installed OSATE 2.2.1 to 2.2.2 using the update site? That is not guaranteed to work.

Did you test on a fresh workspace? A full build is necessary on an existing workspace because of the meta-model changes (to recreate the content of the .aadlbin-gen directory). Instance models must be regenerated as well.

Can you reproduce the issue if you use the sources from tag 2.2.2-RELEASE -20170515? I just pushed the tags to all repos.

**Comment by Etienne13** [comment]
Hi Lutz,

I finally found where my problem came from: when uninstalling OSATE+BA, it seems that the binaries of the BA were not removed. Then, when re-installing OSATE+BA the binaries of the BA were not replaced by their new versions, while new binaries for OSATE were installed. This led to inconsistencies.

I think that a way to avoid this problem would be to rename plugins qualifiers even though the source code did not change. This way, plugins get updated all together. This is just a suggestion, I did not test it.

Thanks for your inputs,
Etienne.
